src/amber.cc

```diff
@@ -51,39 +51,52 @@ amber::Result Amber::Parse(const std::string& input, amber::Recipe* recipe) {
   return {};
 }
 
+namespace {
+
 // Create an engine initialize it, and check the recipe's requirements.
 // Returns a failing result if anything fails.  Otherwise pass the created
-// engine out through |engine_ptr| and the script via |script|.
-static Result CreateEngineAndCheckRequirements(
-    const Recipe* recipe,
-    Options* opts,
-    std::unique_ptr<Engine>* engine_ptr,
-    Script** script_ptr) {
+// engine out through |engine_ptr| and the script via |script|.  The |script|
+// pointer is borrowed, and should not be freed.
+Result CreateEngineAndCheckRequirements(const Recipe* recipe,
+                                        Options* opts,
+                                        std::unique_ptr<Engine>* engine_ptr,
+                                        Script** script_ptr) {
   if (!recipe)
     return Result("Attempting to check an invalid recipe");
 
   Script* script = static_cast<Script*>(recipe->GetImpl());
   if (!script)
     return Result("Recipe must contain a parsed script");
 
+  const auto saved_env = script->GetSpvTargetEnv();
```

Comment:
What is this for? Do you expect the environment to change somehow? This seems like a complicated way of just calling script->SetSpvTargetEnv(saved_env) at the various usage sites. ...

Reply:
I wanted to avoid having to mention anything about the mutation of the recipe.
Without this, the recipe's script actually changes state, which is potentially surprising.   I will avod the saving and instead enforce setting in the usage sites.

Reply:
I think we can just add a note in the comment and allow it for now. I don't think it will cause an issue at the moment.